### ISSUE TITLE
Combining Messages in a Records

### ISSUE THREAD

**Issue by oreizi** [tls-attacker/TLS-Attacker#160]
Is it possible to have the ClientKeyExchange, ChangeCipherSpec and Finished messages all sent in the same Record? It seems like with the default configuration, these three TLS messages are split into three spearate Records (TCP packets). What I have tried so far is adding the:
```
        <records>
            <Record>
            </Record>
        </records>
```
to the SendAction but this did not result in the wanted behaviour.

I tried as well to set the configuration bool sendHandshakeMessagesWithinSingleRecord but had no luck with it as well.

Tested with TLS-Attacker version: 5.3.0

**Comment by ic0ns** [comment]
Just to give some quick feedback: with TLS-Attacker 5.3.0 I think this is not (easily) possible anymore. _sendHandshakeMessagesWithinSingleRecord_ only controls handshake message fragmentation. Since the changecipherspec message is not a handshake message TLS-Attacker will split the message. However, in previous versions of TLS-Attacker we had a dedicated flag for exactly this case called  _flushOnMessageTypeChange_ in TLS-Attacker 4.0.2. The reason we removed this nativ feature is that the record will be effectively not be parseable for the peer - so we did not see any reason to keep it - but maybe you will find it helpful.
